### PR TITLE
feat: Add additional variants for Course Details and Introduce Your C…

### PR DESCRIPTION
**Pull request by ruzniaievdm**
[2U-111](https://youtrack.raccoongang.com/issue/2U-111)
[2U-194](https://youtrack.raccoongang.com/issue/2U-194)
[2U-196](https://youtrack.raccoongang.com/issue/2U-196)
[2U-197](https://youtrack.raccoongang.com/issue/2U-197)
[2U-205](https://youtrack.raccoongang.com/issue/2U-205)
[2U-198](https://youtrack.raccoongang.com/issue/2U-198)

### CODE REVIEW

File: src/generic/course-upload-image/index.jsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  const cardImageTextBody = assetImagePath ? (
+    <span className="x-small text-gray-700">
+      <FormattedMessage
+        {...messages.uploadImageBodyFilled}
+        values={{
+          hyperlink: (
+            <Hyperlink
+              destination={assetsUrl.href}
+              target="_blank"
+              rel="noopener noreferrer"
```
Comment (PKulkoRaccoonGang) [inline src/generic/course-upload-image/index.jsx:93]: Paragon Hyperlink component has `rel="noopener noreferrer"` by default. Please, remove this line.

File: src/generic/course-upload-image/messages.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  uploadImageFilesAndUploads: {
+    id: 'course-authoring.schedule-section.introducing.upload-image.file-and-uploads',
+    defaultMessage: 'files and uploads',
+  },
+  uploadImageDropzoneText: {
+    id: 'course-authoring.schedule-section.introducing.upload-image.dropzone-text',
+    defaultMessage: 'Drag and drop your {identifierFieldText} here or click to upload.',
+  },
+  uploadImageDropzoneAlt: {
+    id: 'course-authoring.schedule-section.introducing.upload-image.dropzone-text',
```
Comment (PKulkoRaccoonGang) [inline src/generic/course-upload-image/messages.js:17]: This identifier is a duplicate, please rename

File: src/generic/help-sidebar/index.jsx [hunk 3]
```diff
@@ -67,22 +69,22 @@ const HelpSidebar = ({
...
                     {intl.formatMessage(messages.sidebarLinkToCourseTeam)}
                   </Hyperlink>
                 </li>
               )}
               {proctoredExamSettingsUrl
-                && pathname !== proctoredExamSettingsUrl && (
+                && !proctoredExamSettingsUrl.includes(pathname) && (
                 <li className="help-sidebar-other-link">
                   <Hyperlink
                     rel="noopener"
```
Comment (PKulkoRaccoonGang) [inline src/generic/help-sidebar/index.jsx:90]: Paragon Hyperlink component has `rel="noopener noreferrer"` by default. Please remove `rel="noreferrer"` everywhere in this file

File: src/schedule-and-details/basic-section/index.jsx [hunk 4]
```diff
@@ -96,7 +93,11 @@ const BasicSection = ({
 
   const renderCoursePromotion = () => (
     <Card>
-      <Card.Header className="h4 px-3 text-gray-500" title={promotionTitle} size="sm" />
+      <Card.Header
+        className="h4 px-3 text-gray-500"
+        title={promotionTitle}
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/basic-section/index.jsx:98]: Should the `promotionTitle` be wrapped in a translation?

File: src/schedule-and-details/instructors-section/InstructorsSection.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.instructors-section {
+  .instructors-list {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/InstructorsSection.scss:2]: `.instructors-section .instructors-list { ... }`

File: src/schedule-and-details/instructors-section/InstructorsSection.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    gap: 30px;
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/InstructorsSection.scss:6]: Convert to `rem` please

File: src/schedule-and-details/instructors-section/index.jsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  const intl = useIntl();
+  const newInstructor = {
+    bio: '',
+    image: '',
+    name: '',
+    organization: '',
+    title: '',
+  };
+
+  const handleChange = (value, idx, field) => {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/index.jsx:22]: `updateInstructorField` maybe better

File: src/schedule-and-details/instructors-section/index.jsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      return instructor;
+    });
+
+    const updatedInstructorInfo = {
+      instructors: updatedInstructors,
+    };
+    onChange(updatedInstructorInfo, 'instructorInfo');
+  };
+
+  const handleDelete = (idx) => {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/index.jsx:39]: `deleteInstructorField` as an option for a more meaningful function name

File: src/schedule-and-details/instructors-section/index.jsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    const updatedInstructorInfo = {
+      instructors: [...instructors],
+    };
+    updatedInstructorInfo.instructors.splice(idx, 1);
+    onChange(updatedInstructorInfo, 'instructorInfo');
+  };
+
+  const handleAdd = () => {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/index.jsx:47]: `addInstructorField`

File: src/schedule-and-details/instructors-section/index.jsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { v4 as uuid } from 'uuid';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+import { Add as AddIcon } from '@edx/paragon/icons';
+import InstructorContainer from './instructor-container';
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/index.jsx:7]: please add the empty line and better import ordering

File: src/schedule-and-details/instructors-section/instructor-container/index.jsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TextareaAutosize from 'react-textarea-autosize';
+import {
+  Card, Form, Col, Button,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import CourseUploadImage from '../../../generic/course-upload-image';
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/instructors-section/instructor-container/index.jsx:8]: please add empty line

**Review by sendr — APPROVED** [review]

File: src/schedule-and-details/introducing-section/IntroducingSection.scss [hunk 12]
```diff
@@ -1,2 +1,7 @@
 @import "./introduction-video/IntroductionVideo";
-@import "./course-card-image/CourseCardImage";
+
+.introducing-section {
+  .form-group-custom:not(:last-child) {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/introducing-section/IntroducingSection.scss:4]: `.introducing-section .form-group-custom:not(:last-child) { ... }`

File: src/schedule-and-details/introducing-section/extended-course-details/messages.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import { defineMessages } from '@edx/frontend-platform/i18n';
+
+const messages = defineMessages({
+  extendedTitleLabel: {
+    id: 'course-authoring.schedule-section.introducing.title.label',
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/introducing-section/extended-course-details/messages.js:5]: Id duplication
image

File: src/schedule-and-details/learning-outcomes-section/LearningOutcomes.scss [hunk 14]
```diff
@@ -0,0 +1,10 @@
+.learning-outcomes-section {
+  .learning-outcomes-list {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/learning-outcomes-section/LearningOutcomes.scss:2]: `.learning-outcomes-section .learning-outcomes-list { ... }`

File: src/schedule-and-details/learning-outcomes-section/index.jsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    onChange(updatedInfo, 'learningInfo');
+  };
+
+  const handleDelete = (index) => {
+    const updatedInfo = [...learningInfo];
+    updatedInfo.splice(index, 1);
+    onChange(updatedInfo, 'learningInfo');
+  };
+
+  const handleAdd = () => {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/learning-outcomes-section/index.jsx:25]: `handleAddLearningInfo`

File: src/schedule-and-details/learning-outcomes-section/index.jsx [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+const LearningOutcomesSection = ({ learningInfo, onChange }) => {
+  const intl = useIntl();
+
+  const handleInputChange = (value, index) => {
+    const updatedInfo = [...learningInfo];
+    updatedInfo[index] = value;
+    onChange(updatedInfo, 'learningInfo');
+  };
+
+  const handleDelete = (index) => {
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/learning-outcomes-section/index.jsx:19]: `handleDeleteLearningInfo`

File: src/schedule-and-details/schedule-section/ScheduleSection.scss [hunk 17]
```diff
@@ -2,11 +2,10 @@
   .schedule-date-list {
     display: flex;
     flex-direction: column;
-    padding: 0;
+    padding-top: 1rem;
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/schedule-section/ScheduleSection.scss:5]:
```suggestion
    padding-top: $spacer;
```

File: src/schedule-and-details/basic-section/index.jsx (hunk 4 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/basic-section/index.jsx:98]: we already have it wrapping translate
```
const promotionTitle = (
    <FormattedMessage
      id="course-authoring.schedule.basic.promotion.title"
      defaultMessage="Course summary page {smallText}"
      values={{
        smallText: <small>(for student enrollment and access)</small>,
      }}
    />
  );
```

File: src/generic/help-sidebar/index.jsx (hunk 3 above)
Comment (ruzniaievdm) [reply src/generic/help-sidebar/index.jsx:90]: removed everywhere

File: src/schedule-and-details/instructors-section/InstructorsSection.scss (hunk 6 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/instructors-section/InstructorsSection.scss:6]: translated

File: src/schedule-and-details/instructors-section/index.jsx (hunk 7 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/instructors-section/index.jsx:22]: it is the simplest component with instructor logic, I think `handleChange` would be enough.

File: src/schedule-and-details/introducing-section/extended-course-details/messages.js (hunk 13 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/extended-course-details/messages.js:5]: Good point, thanks! Changed

File: src/schedule-and-details/schedule-section/ScheduleSection.scss (hunk 17 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/schedule-section/ScheduleSection.scss:5]: changed

File: src/generic/course-upload-image/messages.js (hunk 2 above)
Comment (ruzniaievdm) [reply src/generic/course-upload-image/messages.js:17]: Good point, thanks! Changed

File: src/generic/course-upload-image/index.jsx (hunk 1 above)
Comment (ruzniaievdm) [reply src/generic/course-upload-image/index.jsx]: removed

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Pull request merged**